examples/energy-management-app/linux/main.cpp

```diff
@@ -87,22 +98,66 @@ static uint32_t ParseNumber(const char * pString)
 void ApplicationInit()
 {
     ChipLogDetail(AppServer, "Energy Management App: ApplicationInit()");
-    EvseApplicationInit();
+    if (strcmp(spApp, kEvseApp) == 0)
+    {
+        EvseApplicationInit();
+    }
+    else if (strcmp(spApp, kWhmApp) == 0)
+    {
+        FullWhmApplicationInit();
+    }
+    else
+    {
+        ChipLogError(Support, "Unexpected application %s", spApp);
+    }
 }
 
 void ApplicationShutdown()
 {
     ChipLogDetail(AppServer, "Energy Management App: ApplicationShutdown()");
-    EvseApplicationShutdown();
+
+    if (strcmp(spApp, kEvseApp) == 0)
+    {
+        EvseApplicationShutdown();
+    }
+    else if (strcmp(spApp, kWhmApp) == 0)
+    {
+        FullWhmApplicationShutdown();
+    }
+    else
+    {
+        ChipLogError(Support, "Unexpected application %s", spApp);
+    }
 }
 
-static bool FeatureMapOptionHandler(const char * aProgram, chip::ArgParser::OptionSet * aOptions, int aIdentifier,
-                                    const char * aName, const char * aValue)
+static bool EnergyAppOptionHandler(const char * aProgram, chip::ArgParser::OptionSet * aOptions, int aIdentifier,
+                                   const char * aName, const char * aValue)
 {
     bool retval = true;
 
+    ChipLogError(Support, "aIdentifier 0x%04x aValue %s", aIdentifier, aValue);
     switch (aIdentifier)
     {
+    case kOptionApplication:
+        for (uint16_t idx = 0; idx < sizeof(kValidApps) / sizeof(kValidApps[0]); idx++)
```

Comment:
I think there is a macro for sizeof(struct)/sizeof(struct[0]) - is it ARRAY_SIZE?

Reply:
ok